source/DomainTheory/Part-I.lagda

```diff
@@ -0,0 +1,866 @@
+  Proposition-6-7-ad = continuity-of-function (𝓔ᴰ ×ᵈᶜᵖᵒ 𝓓) 𝓔 (eval 𝓓 𝓔)
+
+ open import DomainTheory.Basics.LeastFixedPoint pt fe 𝓥
+
+ module _ (𝓓 : DCPO⊥ {𝓤} {𝓣}) where
+
+  Theorem-6-8 : DCPO[ ((𝓓 ⟹ᵈᶜᵖᵒ⊥ 𝓓) ⁻) , (𝓓 ⁻) ]
+  Theorem-6-8 = μ 𝓓
+
+  Theorem-6-8-i : (f : DCPO[ 𝓓 ⁻ , 𝓓 ⁻ ])
+                → [ (𝓓 ⟹ᵈᶜᵖᵒ⊥ 𝓓) ⁻ , 𝓓 ⁻ ]⟨ μ 𝓓 ⟩ f
+                ＝ [ 𝓓 ⁻ , 𝓓 ⁻ ]⟨ f ⟩ ([ (𝓓 ⟹ᵈᶜᵖᵒ⊥ 𝓓) ⁻ , 𝓓 ⁻ ]⟨ μ 𝓓 ⟩ f)
+  Theorem-6-8-i = μ-gives-a-fixed-point 𝓓
+
+  Theorem-6-8-ii : (f : DCPO[ (𝓓 ⁻) , (𝓓 ⁻) ])
+                   (x : ⟪ 𝓓 ⟫)
+                 → [ 𝓓 ⁻ , 𝓓 ⁻ ]⟨ f ⟩ x  ⊑⟪ 𝓓 ⟫ x
+                 → [ (𝓓 ⟹ᵈᶜᵖᵒ⊥ 𝓓) ⁻ , 𝓓 ⁻ ]⟨ μ 𝓓 ⟩ f ⊑⟪ 𝓓 ⟫ x
+  Theorem-6-8-ii = μ-gives-lowerbound-of-fixed-points 𝓓
+
+\end{code}
+
+Section 7
+
+\begin{code}
+
+module _ (𝓥 : Universe) where
+
+ open import DomainTheory.Basics.Dcpo pt fe 𝓥
+ open import DomainTheory.Basics.Exponential pt fe 𝓥
+ open import DomainTheory.Basics.FunctionComposition pt fe 𝓥
+ open import DomainTheory.Basics.Miscelanea pt fe 𝓥
+
+ Definition-7-1 : (𝓓 : DCPO {𝓤} {𝓣}) → DCPO[ 𝓓 , 𝓓 ] → 𝓤 ⊔ 𝓣 ̇
+ Definition-7-1 = is-deflation
+
+ Definition-7-2 : (𝓓 : DCPO {𝓤} {𝓣}) (𝓔 : DCPO {𝓤'} {𝓣'})
+                → 𝓥 ⁺ ⊔ 𝓤 ⊔ 𝓣 ⊔ 𝓤' ⊔ 𝓣' ̇
+ Definition-7-2 𝓓 𝓔 = Σ s ꞉ DCPO[ 𝓓 , 𝓔 ] ,
+                      Σ r ꞉ DCPO[ 𝓔 , 𝓓 ] , is-embedding-projection-pair 𝓓 𝓔 s r
+
+ module setup
+         {𝓤 𝓣 : Universe}
+         {I : 𝓥 ̇ }
+         (_⊑_ : I → I → 𝓦 ̇ )
+         (⊑-refl : {i : I} → i ⊑ i)
+         (⊑-trans : {i j k : I} → i ⊑ j → j ⊑ k → i ⊑ k)
+         (⊑-prop-valued : (i j : I) → is-prop (i ⊑ j))
+         (I-inhabited : ∥ I ∥)
+         (I-semidirected : (i j : I) → ∃ k ꞉ I , i ⊑ k × j ⊑ k)
+         (𝓓 : I → DCPO {𝓤} {𝓣})
+         (ε : {i j : I} → i ⊑ j → ⟨ 𝓓 i ⟩ → ⟨ 𝓓 j ⟩)
+         (π : {i j : I} → i ⊑ j → ⟨ 𝓓 j ⟩ → ⟨ 𝓓 i ⟩)
+         (επ-deflation : {i j : I} (l : i ⊑ j) (x : ⟨ 𝓓 j ⟩)
+                       → ε l (π l x) ⊑⟨ 𝓓 j ⟩ x )
+         (ε-section-of-π : {i j : I} (l : i ⊑ j) → π l ∘ ε l ∼ id )
+         (ε-is-continuous : {i j : I} (l : i ⊑ j)
+                          → is-continuous (𝓓 i) (𝓓 j) (ε {i} {j} l))
+         (π-is-continuous : {i j : I} (l : i ⊑ j)
+                          → is-continuous (𝓓 j) (𝓓 i) (π {i} {j} l))
+         (ε-id : (i : I ) → ε (⊑-refl {i}) ∼ id)
+         (π-id : (i : I ) → π (⊑-refl {i}) ∼ id)
+         (ε-comp : {i j k : I} (l : i ⊑ j) (m : j ⊑ k)
+                 → ε m ∘ ε l ∼ ε (⊑-trans l m))
+         (π-comp : {i j k : I} (l : i ⊑ j) (m : j ⊑ k)
+                 → π l ∘ π m ∼ π (⊑-trans l m))
+       where
+
+  open import DomainTheory.Bilimits.Directed pt fe 𝓥 𝓤 𝓣
+  open Diagram _⊑_ ⊑-refl ⊑-trans ⊑-prop-valued
+               I-inhabited I-semidirected
+               𝓓 ε π
+               επ-deflation ε-section-of-π
+               ε-is-continuous π-is-continuous
+               ε-id π-id ε-comp π-comp
+  open PosetAxioms
+
+  -- Example-7-3: See DomainTheory.Bilimits.Sequential
```

Comment:
Is this a forgotten link? With a cunning line number.

Reply:
Example 7.3 illustrates the setup of a directed diagram of dcpos and embedding-projection pairs by taking the indexing preorder to be the natural numbers, which is what is worked out in `DomainTheory.Bilimits.Sequential`. But there is not a canonical thing to link to, so I just gave an "informal" link to the file instead.
(The line number is accidental :slightly_smiling_face:)

Reply:
Can you (non-open) import `DomainTheory.Bilimits.Sequential` so that the html readers can click at a link to go there easily?

Reply:
Good idea. I'll do that.